config.py

```diff
@@ -82,194 +102,283 @@
+                self.respath = normpath(join(dirname(sys.executable), pardir, 'Resources'))
+                self.identifier = NSBundle.mainBundle().bundleIdentifier()
 
-            self.default_journal_dir = join(NSSearchPathForDirectoriesInDomains(NSApplicationSupportDirectory, NSUserDomainMask, True)[0], 'Frontier Developments', 'Elite Dangerous')
+            else:
+                self.internal_plugin_dir = join(dirname(__file__), 'plugins')
+                self.respath = dirname(__file__)
+                # Don't use Python's settings if interactive
+                self.identifier = f'uk.org.marginal.{appname.lower()}'
+                NSBundle.mainBundle().infoDictionary()['CFBundleIdentifier'] = self.identifier
 
+            self.default_journal_dir = join(
+                NSSearchPathForDirectoriesInDomains(NSApplicationSupportDirectory, NSUserDomainMask, True)[0],
+                'Frontier Developments',
+                'Elite Dangerous'
+            )
             self.home = expanduser('~')
 
-            self.respath = getattr(sys, 'frozen', False) and normpath(join(dirname(sys.executable), pardir, 'Resources')) or dirname(__file__)
-
-            if not getattr(sys, 'frozen', False):
-                # Don't use Python's settings if interactive
-                self.identifier = 'uk.org.marginal.%s' % appname.lower()
-                NSBundle.mainBundle().infoDictionary()['CFBundleIdentifier'] = self.identifier
-            else:
-                self.identifier = NSBundle.mainBundle().bundleIdentifier()
             self.defaults = NSUserDefaults.standardUserDefaults()
-            self.settings = dict(self.defaults.persistentDomainForName_(self.identifier) or {})	# make writeable
+            self.settings = dict(self.defaults.persistentDomainForName_(self.identifier) or {})  # make writeable
 
             # Check out_dir exists
             if not self.get('outdir') or not isdir(self.get('outdir')):
                 self.set('outdir', NSSearchPathForDirectoriesInDomains(NSDocumentDirectory, NSUserDomainMask, True)[0])
 
-        def get(self, key):
+        def get(self, key: str) -> Union[None, list, str]:
+            """Look up a string configuration value."""
             val = self.settings.get(key)
             if val is None:
                 return None
+
             elif isinstance(val, str):
                 return str(val)
-            elif hasattr(val, '__iter__'):
-                return list(val)	# make writeable
+
+            elif isinstance(val, list):
+                return list(val)  # make writeable
+
             else:
                 return None
 
-        def getint(self, key):
+        def getint(self, key: str) -> int:
+            """Look up an integer configuration value."""
             try:
-                return int(self.settings.get(key, 0))	# should already be int, but check by casting
-            except:
```

Comment:
`ValueError: invalid literal for int() with base 10: 'asd'` et al is the most likely, I believe.

Reply:
I'll add an except for ValueError, but keep the general one for now, just in case.